source/gui/nvdaControls.py

```diff
@@ -77,17 +112,15 @@ def _getPropValue(self, pIDString, dwIDStringLen, idProp):
 				# wx doesn't seem to  have a method to check whether a list item is focused.
 				# Therefore, assume that a selected item is focused,which is the case in single select list boxes.
 				states |= oleacc.STATE_SYSTEM_SELECTED | oleacc.STATE_SYSTEM_FOCUSED
-			return states, 1
+			return states, self.HAS_PROP
 
 class CustomCheckListBox(wx.CheckListBox):
 	"""Custom checkable list to fix a11y bugs in the standard wx checkable list box."""
 
 	def __init__(self, *args, **kwargs):
 		super(CustomCheckListBox, self).__init__(*args, **kwargs)
-		# Import late to prevent circular import.
-		from IAccessibleHandler import accPropServices
 		# Register object with COM to fix accessibility bugs in wx.
-		server = ListCtrlAccPropServer(self)
+		self.server = ListCtrlAccPropServer(self)

```

Comment:
Could this cause one of these circular dependency issues?

Reply:
This represents the ownership and lifetime of the server most accurately. Without this, the relationship and the lifetime of the server is really not clear. It's true that the server holds a weak reference to the control, and something in `SetHwndPropServer` can stop NVDA from being able to exit unless `ClearHwndProps` is called, this is managed by the server binding to the window destroy event.